### PR TITLE
[Mux/Merge] code clean

### PR DESCRIPTION
**Pull request by jaeyun-jung**
code clean, prepare new sync-mode.
- remove unnecessary param
- reduce code complexity
- change return val (true to push buffer to src pad)

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_common.h [hunk 1]
```diff
@@ -144,16 +144,16 @@ gst_tensor_time_sync_get_current_time (GstCollectPads * collect, tensor_time_syn
...
  * It decide which buffer is going to be used according to sync option.
- * @return True / False if EOS, it return TRUE.
+ * @return True to push buffer.
  * @param collect Collect pad.
  * @param sync Synchronization Option (NOSYNC, SLOWEST, BASEPAD, END)
  * @param current_time Current Timestamp
- * @param need_buffer Boolean for Update Collect Pads
  * @param tensors_buf Generated GstBuffer for Collected Buffer
  * @param configs Configuration Info for Collected Buffer
+ * @param is_eos True when EOS (eod-of-stream)
```
Comment (gichan-jang) [inline gst/nnstreamer/tensor_common.h:153]: End?

Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_common.h]: Right!

**Comment by jaeyun-jung** [comment]
2020-06-19 PR updated
- validate config when current caps is set

**Comment by jaeyun-jung** [comment]
2020-06-22 PR udpated
- add internal function to check eos

@helloahn Please check the changes

File: gst/nnstreamer/tensor_common_pipeline.c [hunk 2]
```diff
@@ -95,6 +95,25 @@ gst_tensor_time_sync_set_option_data (tensor_time_sync_data * sync)
   return TRUE;
 }
 
+/**
+ * @brief Internal function to get EOS.
+ */
+static gboolean
+_gst_tensor_time_sync_is_eos (GstCollectPads * collect,
+    tensor_time_sync_data * sync, guint empty)
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_common_pipeline.c:103]: This function requires some explanation: what does "empty" mean?

Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_common_pipeline.c:107]: I updated description about this function.
The param empty is the number of empty pads. (no buffer in pad)

File: gst/nnstreamer/tensor_common_pipeline.c [hunk 3]
```diff
@@ -95,6 +95,29 @@ gst_tensor_time_sync_set_option_data (tensor_time_sync_data * sync)
   return TRUE;
 }
 
+/**
+ * @brief Internal function to detect EOS using the number of empty pads.
+ * @param[in] collect Collect pad.
+ * @param[in] sync Synchronization option.
+ * @param[in] empty The number of empty pads (pad has no buffer).
```
Comment (helloahn) [inline gst/nnstreamer/tensor_common_pipeline.c:102]: looks better

File: gst/nnstreamer/tensor_merge/gsttensormerge.c [hunk 4]
```diff
@@ -786,22 +764,22 @@ gst_tensor_merge_change_state (GstElement * element, GstStateChange transition)
  * @retval FALSE if given input is configured invalid.
  */
 static gboolean
-gst_tensor_merge_set_option_data (GstTensorMerge * filter)
+gst_tensor_merge_set_option_data (GstTensorMerge * tensor_merge)
```
Comment (helloahn) [inline gst/nnstreamer/tensor_merge/gsttensormerge.c:767]: became clearer

**Pull request merged**